game/test_suite/tests/test_attack_controller.py

```diff
@@ -123,20 +137,17 @@ def __init__(self):
         self.assertTrue(self.attacking_bot.has_attacked)
         self.assertFalse(gen1.active)
         self.assertFalse(gen2.active)
+
     def test_generators_unchanged_when_no_avatar_hit(self):
-        # Setup bot position
         self.attacking_bot.position = Vector(0, 1)
 
-        # Generators initially ON
-        class DummyGen:
-            def __init__(self):
-                self.active = True
-
-        gen1 = DummyGen()
-        gen2 = DummyGen()
+        gen1 = self.TestGenerator()
+        gen2 = self.TestGenerator()
+        gen1.activate()
+        gen2.activate()
 
         # Put only a Station (no Avatar) in target tile
-        station = GameObject()
+        station = Station(position=Vector(0, 0))
         board = self.build_board({
             Vector(0, 0): [station],
             Vector(0, 1): [self.attacking_bot]
```

Comment:
instead of doing `board.generators = ...`, you can do this:
```suggestion
            Vector(0, 1): [self.attacking_bot],
            Vector(1, 0): [gen1],
            Vector(1, 1): [gen2],
```